lib/bike_brigade/messaging/slack.ex

```diff
@@ -0,0 +1,21 @@
+defmodule BikeBrigade.Messaging.Slack do
+  alias BikeBrigade.SlackApi
+
+  defmodule RiderSms do
+    @channel_id "C01QU0YVACW" #dispatch
+
+    def post_message(message) do
```

Comment:
I think there's two or three different things that are being complected by `RiderSms` and `Operations` both having the same function with the same argument, and both calling `SlackApi.PayloadBuilder.build` - which decides what the slack message will look like based on the type of the input.

For example, `Operations.post_message(%SmsMessage{...}` will end up posting a message formatted as a rider sms message to #software, and `RiderSms.post_message("hello")` will post "hello" to the dispatch channel.

The naming of these modules makes me thing they are about differentiating the *kind* of slack message we're sending (with also an associated channel). I think it makes sense to move the payload construction to its own module, but maybe the payload builder shouldn't just change based on type?

Reply:
So, it was entirely intentional to _not_ differential the kind of slack messages we send based on the channel name (since we'd maybe want to send different messages with different formatting requirements to the same channel). Here we just use the modules as little helpers to set the correct message destination (other option is doing something like `Slack.message("body", "channel-name")` which is also okay but feels a bit magic-string/atom-y).

Agreed there is a bit of weirdness between the type of message/destination, though I think it's partially because we only have the two use cases at the moment. Thoughts about leaving it as is and returning to it once we have a 3rd use case that might help inform the design?

Reply:
One more thought  - this would be a good place to look into [behaviours](https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#behaviours)